src/vs/workbench/contrib/positronNotebook/browser/notebookCells/NotebookHTMLOutput.tsx

```diff
@@ -71,47 +72,56 @@ function acquireVsCodeApi(): { postMessage: (message: HTMLOutputWebviewMessage)
 
 
 export function NotebookHTMLContent({ content }: { content: string }) {
-	const { webviewService } = useServices();
-
 	const containerRef = React.useRef<HTMLDivElement>(null);
+	const { notebookWebviewService } = useServices();
+	const instance = useNotebookInstance();
+	const notebookRuntime = useObservedValue(instance.currentRuntime);
 
 	React.useEffect(() => {
-		if (!containerRef.current) {
-			return;
-		}
-
-		const webviewElement = webviewService.createWebviewElement({
-			title: localize('positron.notebook.webview', "Positron Notebook HTML content"),
-			options: {
-				purpose: WebviewContentPurpose.NotebookRenderer,
-				enableFindWidget: false,
-				transformCssVariables: transformWebviewThemeVars,
-			},
-			contentOptions: {
-				allowMultipleAPIAcquire: true,
-				allowScripts: true,
-			},
-			extension: undefined,
-			providedViewType: 'notebook.output'
-		});
-
-		const contentWithStyles = buildWebviewHTML({
-			content,
-			styles: htmlOutputStyles,
-			script: `(${webviewMessageCode.toString()})();`
-		});
-
-		webviewElement.setHtml(contentWithStyles);
-		webviewElement.onMessage(({ message }) => {
-			if (!isHTMLOutputWebviewMessage(message) || !containerRef.current) { return; }
-			// Set the height of the webview to the height of the content
-			// Don't allow the webview to be taller than 1000px
-			const boundedHeight = Math.min(message.bodyScrollHeight, 1000);
-			containerRef.current.style.height = `${boundedHeight}px`;
-		});
-		webviewElement.mountTo(containerRef.current, getWindow(containerRef.current));
-		return () => webviewElement.dispose();
-	}, [content, webviewService]);
+		const containerElement = containerRef.current;
+		if (!containerElement) { return; }
+
+		let disposed = false;
+		// Cleanup function that will be overwritten if the webview is created.
+		// If the effect gets cleaned up before the webview has been rendered it will
+		// set the disposed variable to true letting the webview creation know not to
+		// mount the webview.
+		let cleanup = () => { disposed = true; };
+
+		const buildWebview = async () => {
+			const webviewElement = await notebookWebviewService.createRawHtmlOutput({
+				id: 'notebook-html-output',
+				runtimeOrSessionId: notebookRuntime ?? 'positron.notebooks.output.render',
```

Comment:
I wonder if these should be unique? Might avoid issues down the line. In the console case both are UUIDs (I think).

Reply:
Oh I didnt realize that. I assumed that a runtime could have multiple webviews associated. I can make it unique!

Reply:
Now the id of the webview is the cell's `outputId` and the session id is a unique ID for the notebook instance when the runtime has not yet started up.